doc/sfp-cmis/CMIS-custom-SI-settings.md

```diff
@@ -0,0 +1,281 @@
+
+### 3.1 TX_SETTING:   
+This section will provide details on whether the TX EQ (TX input equalizer control) setting is FIXED or ADAPTIVE. Only adaptive EQ should be used for TX input, and it's enabled as the default setting in module. Fixed EQ is not recommended for TX direction and will not work until the SI/Hardware team explicitly recommends it.
+
+If the EQ_FIXED flag is false or not present, then the SI param generation flow will come out of TX_SETTING and continue with the RX_SETTING block. But if the EQ_FIXED flag is true for TX_SETTING, then we need to disable AdaptiveInputEqEnableTx.
+
+The TX Input EQ register control: Page 10h Byte 153 – 159 
+
+| **Byte**       | **Field Name**                            |
+| -------------- | ----------------------------------------- |
+| 153            | AdaptiveInputEqEnableTx1..8 (lane 1-8)    |
+| 154 - 155      | AdaptiveInputEqRecallTx1..8 (lane 1-8)    |
+| 156 - 159      | FixedInputEqTargetTx1..8    (lane 1-8)    |
+
+### 3.2 RX_SETTING:   
+The RX_SETTING block contains the same sections as TX_SETTING, but the EQ_FIXED flag should always be true. The SI settings can be directly written and applied for RX output equalization.
+
+The RX Output EQ register control: Page 10h Byte 162 – 173
+
+| **Byte**       | **Field Name**                            |
+| -------------- | ----------------------------------------- |
+| 162 - 165      | OutputEqPreCursorTargetRx1..8 (lane 1-8)  |
+| 166 - 169      | OutputEqPostCursorTargetRx1..8 (lane 1-8) |
+| 170 - 173      | OutputAmplitudeTargetRx1..8    (lane 1-8) |
+
+### 3.3 GLOBAL_MEDIA_SETTINGS:  
+This block's first level of identification will be the range of port numbers. The ports can be defined as a range of 0-31 or a list of multiple ports: 1, 2, 3, or a list of ports in the range of 5–10, 25–31, matching the index number in the port_config.ini file. This port range will have a unique defined lane speed, which will have unique vendor and vendor part number entries supporting this speed. Module key will be created based on speed and vendor details.
+
+Each vendor will have per-lane SI param attribute entries applicable for the identified port + speed for the platform vendor. This value will be searched through the module key.
+
+### 3.4 PORT_MEDIA_SETTINGS:  
+The entries in this block will be unique single port numbers. The control of SI attribute list generation search will reach the PORT_MEDIA_SETTINGS block only when no attribute list is generated in the GLOBAL_MEDIA_SETTINGS block.   
+
+There will be unique speed and vendor/vendor_PN entries in each identified port block.
+
+Default values can be platform defaults for multiple vendors in each section.
+
+### 3.5 List of standard TX RX SI parameters
+-  FixedInputEqTargetTx 
+-  OutputEqPreCursorTargetRx 
+-  OutputEqPostCursorTargetRx 
+-  OutputAmplitudeTargetRx
+
+### 3.6 Sample Optics SI setting file:
+```
+{
+        "TX_SETTING": {
+                "EQ_FIXED": "False",
```

Comment:
@AnoopKamath why do you assume the TX Eq will be fixed? Should we not check module advt. on Page 01h, Byte 161 bit 3?

Reply:
Removed TX/RX/EQ_FIXED details from JSON. I had added TX EQ details in JSON file to start with. I am checking this in code before applying. TX/RX details can be added directly in the JSON